test/composition.cancel.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const metasync = require('..');
+const metatests = require('metatests');
+
+
+// Emulate Asynchronous calls of function
+//   callback - function
+// Result: undefined; call of function
+const wrapAsync = callback => {
+  setTimeout(callback, Math.floor(Math.random() * 500));
+};
+
+metatests.test('async functions composition cancel in the middle', test => {
+
+  let fc = null;
+  const fn1 = (data, cb) => {
```

Comment:
use `test.mustCall` and `test.mustNotCall` where the `fn`s must (or must not) be called.
i.e.

```js
const fn1 = test.mustCall((data, cb) => {
 ...
});